### ISSUE TITLE
Solving a system with RESULT = input matrix

### ISSUE THREAD

**Issue by whuyse** [zathras/jrpn#84]
**JRPN 15C**

**Describe the bug**
Solving a system A*X=B (X is any matrix but A) is done with
RCL MATRIX B
RCL MATRIX A
RESULT X
/
This does not work when X equals B and - as far as I can determine - A contains pivot information.

**Debug Log**
"Next, go to Menu -> Calculator State -> State with Comments -> Copy to Clipboard. "
That option is not in the menu

**To Reproduce**
run this program:
```
#  Program produced by JRPN 15C.
#  Generated 2024-3-13 10:24 Central European Standard Time.

   000 {          } 
   001 { 42 21 11 } f LBL A
   002 { 42 16  0 } f MATRIX 0
   003 {        2 } 2
   004 {       36 } ENTER
   005 { 42 23 11 } f DIM A
   006 {        1 } 1
   007 { 42 23 12 } f DIM B
   008 { 42 16  1 } f MATRIX 1
   009 {    44 12 } STO B
   010 {        2 } 2
   011 {  44 11 u } u STO A
   012 {        1 } 1
   013 {  44 11 u } u STO A
   014 {        4 } 4
   015 {  44 11 u } u STO A
   016 {        2 } 2
   017 {       16 } CHS
   018 {  44 11 u } u STO A
   019 {       34 } X<=>Y
   020 { 45 16 12 } RCL MATRIX B
   021 { 45 16 11 } RCL MATRIX A
   022 { 42 26 12 } f RESULT B
   023 {       10 } /
   024 {    43 32 } g RTN

# End.
```

**Expected behavior**
We're solving 
```
[ 2 1] [ x ] = [ 1 ]
[4 -2] [ y ]   [ 0 ]
```
matrix B now should contain [0.25 0.5], but it contains [0 0]
if you use RESULT C, it is correct

**Platform:**
JRPN 15C Version 2.1.12beta2 web running on Windows 10 Enterprise under Chrome 122.0.6261.112

**Additional context**

**Comment by zathras** [comment]
Thanks for the detailed writeup!  That program will definitely go in the regression test suite once I fix this.

**Comment by zathras** [comment]
Published in 2.1.12
